brainscore_language/benchmarks/futrell2018_gam/benchmark.py

```diff
@@ -0,0 +1,132 @@
+    Alignment of reading times between model and human subjects is evaluated via a generalized additive
+    linear model, incorporating current- and previous-word surprisals along with control properties
+    of word length and word frequency.
+    """
+
+    FORMULA = "reading_time ~ s(surprisal, bs='cr', k=20) + s(prev_surp, bs='cr', k=20) + " + \
+              "te(freq, len, bs='cr') + te(prev_freq, prev_len, bs='cr')"
+
+    def __init__(self):
+        self.data = load_dataset('Futrell2018')
+        self.metric = load_metric('pearsonr')
+        ceiler = SplitHalvesConsistency(num_splits=10, split_coordinate='subject_id', consistency_metric=self.metric)
+        ceiling = ceiler(self.data)
+
+        # Load R dependencies.
+        numpy2ri.activate()
+        pandas2ri.activate()
+
+        super(Futrell2018GAMPearsonr, self).__init__(
+            identifier='Futrell2018-GAM-pearsonr',
+            version=1,
+            parent='behavior',
+            ceiling=ceiling,
+            bibtex=self.data.bibtex)
+
+    def fit(self, surprisals, data_mask):
+        formula = ro.Formula(self.FORMULA)
+
+        data = pd.DataFrame({
+            "surprisal": surprisals,
+            "reading_time": self.data[data_mask].mean("subject"),
+        })
+        data["prev_surp"] = data["surprisal"].shift(1)
+        data["len"] = self.data[data_mask].word_core.str.len()
+        data["prev_len"] = data["len"].shift(1)
+        data["freq"] = surprisals  # HACK need to look this up.
+        data["prev_freq"] = data["freq"].shift(1)
+
+        # Second round of masking, excluding for which there are nan values (e.g. first word has no defined prev features)
+        data_mask = ~data.isna().any(axis=1)
+        data = data[data_mask]
+
+        # TODO check that columns match formula variable names
+
+        r_mgcv = importr("mgcv")
+        model = r_mgcv.gam(formula, data=data)
+
+        # TODO held out data
+        predictions = r_mgcv.predict_gam(model, newdata=data, type="response")
+
+        return model, predictions, data.reading_time
+
+    def __call__(self, candidate: ArtificialSubject) -> Score:
+        # run experiment
+        candidate.start_behavioral_task(ArtificialSubject.Task.reading_times)
+        stimuli = self.data['word'].values
+        surprisals = candidate.digest_text(stimuli)['behavior']
+        attach_presentation_meta(surprisals, self.data['presentation'])
+
+        # exclude first words
+        surprisals = surprisals[surprisals['word_within_sentence_id'] != 1] 
+        data_mask = self.data['word_within_sentence_id'] != 1
+
+        # Fit and evaluate GAM model
+        model, predictions, targets = self.fit(surprisals, data_mask)
+
+        # score
+        raw_score = self.metric(predictions, targets)
+        score = ceiling_normalize(raw_score, self.ceiling)
+        return score
+
+
+class SplitHalvesConsistency:
```

Comment:
could `from ../futrell2018.benchmark import SplitHalvesConsistency` (https://github.com/brain-score/language/blob/01b229ce713fe90a029d3742ba24653a64ea6aa7/brainscore_language/benchmarks/futrell2018/benchmark.py#L55) since identical. Or we put both benchmarks inside the `benchmarks/futrell2018` plugin? I'm fine with either, slightly leaning towards adding this to the `futrell2018` plugin